### ISSUE TITLE
OSWindows host on Windows: a segmentation fault after mouse click

### ISSUE THREAD

**Issue by tinchodias** [pharo-graphics/Bloc#122]
Download Pharo 90+vmHeadlessLatest, load Bloc and Memory game (https://github.com/pharo-graphics/Tutorials) and finally open the "space" using the BlOSWindowSDL2Host host (see below). Now, click on a card to crash.

Error:
```
./pharo-ui: line 13:   556 Segmentation fault      "$DIR"/"pharo-vm/Pharo.exe" "$image_name" --interactive "$@"
```
I don't have a crash.dmp or PharoDebug.log.

Code:
```
aGameElement := MGGameElement new
	memoryGame: MGGame withNumbers;
	yourself.
space := BlSpace new.
space host: BlOSWindowSDL2Host new.
space addChild: aGameElement. 
space show. 
```

**Comment by tinchodias** [comment]
For the record, Pablo suggested me to launch Pharo on Windows this way to have extended logging: `pharo-vm/PharoConsole.exe --logLevel=4 Pharo.image --interactive`

This has 2 points to use:
- PharoConsole.exe
- logLevel=4

(I think vm needs to be vmHeadlessLatest, currently)

**Comment by tinchodias** [comment]
I attach the output after clicking on a "memory card".

[output.txt](https://github.com/pharo-graphics/Bloc/files/6207441/output.txt)

**Comment by tinchodias** [comment]
For the record, yesterday we had a debugging session with Pablo but it didn't lead us to a code to fix. The conjecture now is that there is something wrong in `BlOSWindowSDL2Host` and collaborators such as a double free of some object. That would put the stack/heap in a inconsistent state that leads to a posterior error/segmention fault that isn't connected to the origin of the problem (e.g. the double free). So, I will do a pass reading the code.

**Comment by tinchodias** [comment]
A more minimal snippet to reproduce the Segmentation Fault in Window. It happens with both alternatives, Skia and Cairo canvas:
```smalltalk
Bloc preferableHostClass: BlOSWindowSDL2Host.
Bloc preferableSpartaCanvas: BlSpartaSkiaCanvasBuilder.
"Bloc preferableSpartaCanvas: BlSpartaCairoCanvasBuilder."

space := BlSpace new; show; yourself.
"
space root invalidate.
"
```

**Comment by tinchodias** [comment]
For the record, two notes:

# 1
Both skia and cairo canvas make Pharo 10 crash in Windows, and **almost always** it crashes after the same FFI class, but in few cases it was an other functions, which for the moment confirms the hypothesis of Pablo that something is wrong in memory (drafted 2 comments above).

Almost always, the Segmentation Fault in Windows is at:
- _pixman_implementation_create_mmx function (libpixman-1-0.dll module) for CAIRO
- rust_eh_personality function (libSkia.dll module) for SKIA

---

# 2

These are captures of the calls done via ThreadedFFI in Pharo10+Macos (no segfault here), each time `invalidate` is sent to the root element (see example in previous comment):

<img width="1087" alt="SkiaMac" src="https://user-images.githubusercontent.com/3044265/129645002-210d4415-0b27-48cf-9dea-173975e945f4.png">
<img width="1090" alt="CairoMac" src="https://user-images.githubusercontent.com/3044265/129644994-2b91216b-c937-483a-a0f4-72838a73c132.png">

**Comment by tinchodias** [comment]
Fixed with 19517e64c594f6035075a580eb5c3ab625500ed4
